File: backend/src/models/Correction.js
'use strict';

const mongoose = require('mongoose');
const { Schema } = mongoose;

const correctionSchema = new Schema(
  {
    messageId: { type: Schema.Types.ObjectId, ref: 'Message', required: true, index: true },
    original: { type: String, required: true },
    corrected: { type: String, required: true },
    reason: { type: String, default: '' },
    example: { type: String, default: '' },
    difficultyLevel: {
      type: String,
      enum: ['beginner', 'intermediate', 'advanced'],
      default: 'beginner',
    },
  },
  { timestamps: { createdAt: true, updatedAt: false } }
);

module.exports = mongoose.model('Correction', correctionSchema);

File: android/app/src/main/java/com/englishfriendai/app/domain/model/Correction.kt
package com.englishfriendai.app.domain.model

enum class DifficultyLevel { BEGINNER, INTERMEDIATE, ADVANCED }

data class Correction(
    val original: String,
    val corrected: String,
    val reason: String,
    val example: String,
    val difficultyLevel: DifficultyLevel
)
